File: solution/2400-2499/2485.Find the Pivot Integer/Solution.cpp
class Solution {
public:
    int pivotInteger(int n) {
        for (int x = 1; x < 1000; ++x) {
            if ((1 + x) * x == (x + n) * (n - x + 1)) {
                return x;
            }
        }
        return -1;
    }
};

File: solution/2400-2499/2485.Find the Pivot Integer/Solution.php
class Solution {
    /**
     * @param Integer $n
     * @return Integer
     */
    function pivotInteger($n) {
        $sum = ($n * ($n + 1)) / 2;
        $pre = 0;
        for ($i = 1; $i <= $n; $i++) {
            if ($pre + $i === $sum - $pre) {
                return $i;
            }
            $pre += $i;
        }
        return -1;
    }
}